src/durablefailure.ts

```diff
@@ -0,0 +1,5 @@
+export class DurableFailure extends Error {
```

Comment:
Would the name `DurableError` make more sense this this class derives from `Error`? Just like how in .NET exception types should end with `Exception` I wonder if we should be doing the same for custom error types in TypeScript. It would make the usage of this type more immediately obvious to other TypeScript/JavaScript developers.